specs/tree-view/tree-view.md

```diff
@@ -0,0 +1,187 @@
+  
+### Features
+
+- **Selection:** A tree view can support either single select or multi-select behavior for nodes. **I would propose that for the initial spec we start with supporting single select only.*
+
+
+- **Drag & Drop support:** Often, as is in the case of a "file system" example, drag and drop support is desired. This does present [challenges](#drag-&-drop).
+
+### Risks and Challenges
+
+#### Nesting:
+
+Infinite nesting combined with the current visual design presents a styling challenge. Consider a scenario where we have the following tree:
+```html
+<fast-tree-view>
+    <fast-tree-item>Root node one</fast-tree-item>
+    <fast-tree-item>Root node two</fast-tree-item>
+    <fast-tree-item>
+        Parent Node && Root node
+        <fast-tree-item slot="item">Nested node one</fast-tree-item slot="item">
+        <fast-tree-item slot="item">Nested node two</fast-tree-item>
+    </fast-tree-item>
+    <fast-tree-item>Root node three</fast-tree-item>
+</fast-tree-view>
+```
+In this scenario, there is no root node with an expand/collapse glyph. The third item in the tree view is a parent node, which *would* have an expand/collapse glyph to the right. To ensure that the content of the parent node align with the rest of the root nodes, the entire tree will need to visually shift to ensure alignmnet of the tree content. This presents an interesting challenge for how we will communicate these changes to the parent to ensure the styling is appropriately applied. Ultimately, the parent just needs to know that *one* of the items below has child nodes.
+
+#### Drag & Drop
+
+Drag & Drop support poses a unique set of challenges, especially if our goal is to provide out of the box support via the [HTML Drag and Drop API](https://developer.mozilla.org/en-US/docs/Web/API/HTML_Drag_and_Drop_API). Alternatively, those using our components with React, Vue, Angular, etc may want to use libraries built to work with their specific framework flavor. 
+
+**Considering the challenge that drag & drop would present for any given component, I think it's worth holding on that feature, and taking it up seperately.*
+
+### Prior Art/Examples
+- [FAST-DNA (React)](https://explore.fast.design/components/tree-view)
+- [Material UI](https://material-ui.com/components/tree-view/)
+- [Lightning Design](https://www.lightningdesignsystem.com/components/trees/)
+- [Ant Design](https://ant.design/components/tree/)
+- [Atlassian](https://atlaskit.atlassian.com/packages/core/tree/example/static-tree)
+- [Windows (UWP)](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/tree-view)
+
+---
+
+## Design
+
+### API
+
+*The key elements of the component's public API surface:*
+
+**Tree View**
+
+- extends HTML Element props
+- custom CSS property for `depth` to support the infinite nesting scenario
+
+**Tree View Item**
+
+- expanded - nodes with child nodes (tree items) can either be expanded or collapsed. 
```

Comment:
Can you categorize these in a similar way that @eljefe223 did? I think it adds clarity to organize by properties, attributes, events, and slots

Reply:
Sure - can we add these preferences to the template itself?

Reply:
nvm, I see how this is inferred